__tests__/Unit/Components/Tasks/TaskDetails.test.tsx

```diff
@@ -1,41 +1,311 @@
+import userEvent from '@testing-library/user-event';
+import TaskDetails, { Button, Textarea } from '@/components/taskDetails';
 import TaskContainer from '@/components/taskDetails/TaskContainer';
 import task from '@/interfaces/task.type';
 import { tasks } from '../../../../__mocks__/db/tasks';
+import { Provider } from 'react-redux';
+import { store } from '@/app/store';
+import { renderWithRouter } from '@/test_utils/createMockRouter';
+import { setupServer } from 'msw/node';
+import handlers from '../../../../__mocks__/handlers';
+import { ButtonProps, TextAreaProps } from '@/interfaces/taskDetails.type';
+import { ToastContainer } from 'react-toastify';
+import Details from '@/components/taskDetails/Details';
 
 const details = {
     url: 'https://realdevsquad.com/tasks/6KhcLU3yr45dzjQIVm0J/details',
     taskID: '6KhcLU3yr45dzjQIVm0J',
 };
 
-describe.skip('TaskDetails Page', () => {
-    test('Loading text rendered when loading', () => {
-        render(<TaskDetails url={details.url} taskID={details.taskID} />);
-        const loadingElement = screen.getByText(/Loading.../i);
-        expect(loadingElement).toBeInTheDocument();
+const server = setupServer(...handlers);
+
+beforeAll(() => {
+    server.listen({
+        onUnhandledRequest: 'error',
     });
-    test('Task title is Editable in Editing mode ', () => {
-        render(<TaskDetails url={details.url} taskID={details.taskID} />);
+});
+afterEach(() => server.resetHandlers());
+afterAll(() => server.close());
 
-        const titleElement = screen.queryByTestId('task-title');
-        expect(titleElement).not.toBeInTheDocument();
+jest.mock('@/hooks/useUserData', () => {
+    return () => ({
+        data: {
+            roles: {
+                admin: true,
+                super_user: false,
+            },
+        },
+        isUserAuthorized: true,
+        isSuccess: true,
     });
-    test('Edit button is not rendered when Editing', () => {
-        render(<TaskDetails url={details.url} taskID={details.taskID} />);
+});
 
-        const editButtonElement = screen.queryByRole('button', {
-            name: 'Edit',
+describe('TaskDetails Page', () => {
+    it('Should render title', async () => {
+        const { getByText } = renderWithRouter(
+            <Provider store={store()}>
+                <TaskDetails taskID={details.taskID} />
+            </Provider>
+        );
+        await waitFor(() => {
+            expect(getByText('test 1 for drag and drop')).toBeInTheDocument();
         });
-        expect(editButtonElement).not.toBeInTheDocument();
     });
-    test('Task Description is Editable in Editing mode', () => {
-        render(<TaskDetails url={details.url} taskID={details.taskID} />);
 
-        const descriptionElement = screen.queryByText(
-            /No description available/i
+    it('should render update progress button ', async () => {
+        const { getByText } = renderWithRouter(
+            <Provider store={store()}>
```

Comment:
I think we can use beforeEach for `renderWithRouter` function here, and as for `getByText` you can use `screen.getByText`

Reply:
Here in the test-suite I am testing few components which don't need the parent component to be rendered before, So I have rendered it without using beforeEach